Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py

```diff
@@ -484,60 +664,66 @@ def main() -> None:
     params = demisto.params()
     command = demisto.command()
     args = demisto.args()
+    demisto.debug(f'{params=}\n{args=}')
 
     demisto.debug(f'Command being called is {command}')
     try:
-        client = Client(
-            tenant_id=params.get('tenant_id', ''),
-            auth_type=params.get('auth_type', 'Device Code'),
-            auth_code=params.get('auth_code', {}).get('password', ''),
-            redirect_uri=params.get('redirect_uri', ''),
-            enc_key=params.get('credentials', {}).get('password', ''),
-            app_id=params.get('app_id', ''),
-            subscription_id=params.get('subscription_id', ''),
-            resource_group_name=params.get('resource_group_name', ''),
-            verify=not params.get('insecure', False),
-            proxy=params.get('proxy', False),
-            azure_ad_endpoint=params.get('azure_ad_endpoint',
-                                         'https://login.microsoftonline.com') or 'https://login.microsoftonline.com',
-            managed_identities_client_id=get_azure_managed_identities_client_id(params)
-        )
+        subscription_id = args.get('subscription_id') or params.get('subscription_id', '')
+        resource_group_name = args.get('resource_group_name') or params.get('resource_group_name', '')
+        commands_with_multiple_resource_group_name_list = ['azure-sql-servers-list',
+                                                           'azure-sql-db-audit-policy-list',
+                                                           'azure-sql-db-audit-policy-create-update',
+                                                           'azure-sql-db-threat-policy-create-update']
+
         if command == 'test-module':
+            client = create_client(params, subscription_id, resource_group_name)
```

Comment:
Why not call this once initially and then pass the client to the functions? 
Instead of creating the client for each command separately